swarm-private/templates/smoke.cronjob.yaml

```diff
@@ -8,30 +8,45 @@ metadata:
   name: {{ template "swarm.fullname" $root }}-smoke
   labels:
     app: {{ template "swarm.name" $root }}
+    chart: {{ template "swarm.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
 spec:
   schedule: "{{ $root.Values.smoke.schedule }}"
+  successfulJobsHistoryLimit: {{ $root.Values.smoke.successfulJobsHistoryLimit }}
+  failedJobsHistoryLimit: {{ $root.Values.smoke.failedJobsHistoryLimit }}
+  concurrencyPolicy: {{ $root.Values.smoke.concurrencyPolicy | quote }}
   jobTemplate:
     spec:
       template:
         spec:
           containers:
           - name: smoke
-            image: ethdevops/swarm-smoke:latest
-            imagePullPolicy: Always
+            image: {{ .Values.smoke.image.repository }}:{{ .Values.smoke.image.tag }}
+            imagePullPolicy: {{ .Values.smoke.imagePullPolicy | quote }}
+            resources:
+{{ toYaml .Values.swarm.resources | indent 14 }}
+            env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
             args:
             - --cluster-endpoint={{ $root.Release.Namespace }}
             - --app={{ template "swarm.fullname" $root }}
             - --cluster-from=0
             - --cluster-to={{ $root.Values.swarm.replicaCount }}
             - --cluster-scheme=http
-            - --filesize={{ .Values.smoke.size }}
-            - --verbosity=5
+            - --filesize={{ .Values.smoke.config.filesize }}
+            - --verbosity={{ .Values.smoke.config.verbosity }}
+        {{- if .Values.swarm.metricsEnabled }}
```

Comment:
I see your point. In that case, it would probably make sense to split the smoke tests to a separate chart that could that be deployed how many times you want with different configurations. e.g. targeting different clusters, using different filsizes.. etc.

For now, I've removed that condition.